pre_push.py

```diff
@@ -41,11 +50,67 @@ def main():
     finally:
         rmtree(tmp_dir)
 
-    return 0 if success else 1
+    return success
+
+
+def run_unit():
+    """Runs the unit-tests.
+
+    Follows the behavior of the static tests,
+    where any failed tests cause pre_push.py to fail.
+    """
+    curdir = path.abspath(path.join(__file__, ".."))
+    return do_process([sys.executable, curdir + "/setup.py", "test"])
```

Comment:
Can you use `"".format` here rather than string concatenation? I'm actually more partial to f-strings now, however, python 3.5 doesn't support them, and it's not EOL until September 2020.

Reply:
I will change it to format strings rather than string concats, but I prefer string concats because all I need to concat a string is % rather than .format